tests/trestle/tasks/csv_to_oscal_cd_test.py

```diff
@@ -268,10 +276,361 @@ def _validate3(tmp_path: pathlib.Path):
+    assert len(cd.components[0].props) == 62
+    assert len(cd.components[0].control_implementations[0].implemented_requirements) == 21
+    assert len(cd.components[0].control_implementations[0].set_parameters) == 4
+
+
+def _validate13(tmp_path: pathlib.Path):
+    # read catalog
+    fp = pathlib.Path(tmp_path) / 'component-definition.json'
+    cd = ComponentDefinition.oscal_read(fp)
+    # spot check
+    assert len(cd.components) == 1
+    assert len(cd.components[0].control_implementations) == 1
+    assert len(cd.components[0].props) == 36
+    assert len(cd.components[0].control_implementations[0].implemented_requirements) == 21
+    assert len(cd.components[0].control_implementations[0].set_parameters) == 2
+
+
+def _validate14(tmp_path: pathlib.Path):
+    # read catalog
+    fp = pathlib.Path(tmp_path) / 'component-definition.json'
+    cd = ComponentDefinition.oscal_read(fp)
+    # spot check
+    assert len(cd.components) == 1
+    assert len(cd.components[0].control_implementations) == 1
+    assert len(cd.components[0].props) == 52
+    assert len(cd.components[0].control_implementations[0].implemented_requirements) == 13
+    assert len(cd.components[0].control_implementations[0].set_parameters) == 4
+
+
+def _validate15(tmp_path: pathlib.Path):
+    # read catalog
+    fp = pathlib.Path(tmp_path) / 'component-definition.json'
+    cd = ComponentDefinition.oscal_read(fp)
+    # spot check
+    assert len(cd.components) == 1
+    assert len(cd.components[0].control_implementations) == 1
+    assert len(cd.components[0].props) == 62
+    assert len(cd.components[0].control_implementations[0].implemented_requirements) == 21
+    assert len(cd.components[0].control_implementations[0].set_parameters) == 4
+    assert 'MODIFIED' in cd.components[0].props[1].value
+
+
+def _validate16(tmp_path: pathlib.Path):
+    # read catalog
+    fp = pathlib.Path(tmp_path) / 'component-definition.json'
+    cd = ComponentDefinition.oscal_read(fp)
+    # spot check
+    assert len(cd.components) == 1
+    assert len(cd.components[0].control_implementations) == 1
+    assert len(cd.components[0].props) == 64
+    assert len(cd.components[0].control_implementations[0].implemented_requirements) == 21
+    assert len(cd.components[0].control_implementations[0].set_parameters) == 4
+    assert 'ADD' in cd.components[0].props[4].value
+    assert 'ADD' in cd.components[0].props[5].value
+
+
+def _validate17(tmp_path: pathlib.Path):
+    # read catalog
+    fp = pathlib.Path(tmp_path) / 'component-definition.json'
+    cd = ComponentDefinition.oscal_read(fp)
+    # spot check
+    assert len(cd.components) == 1
+    assert len(cd.components[0].control_implementations) == 1
+    assert len(cd.components[0].props) == 60
+    assert len(cd.components[0].control_implementations[0].implemented_requirements) == 21
+    assert len(cd.components[0].control_implementations[0].set_parameters) == 4
+
+
+def _validate18(tmp_path: pathlib.Path):
```

Comment:
validate 10,11,12,13,14,17,18 contain identical code except the number of different components. Please rewrite this.
 The same applies to all other `validateXX` functions. They are all highly similar and can be easily compressed...
 
 You can either use one function that takes those as parameters or use parameterized test (via `@pytest.mark.parametrize`)